### DISCUSSION TITLE
Enhance "Transformations / Add field from calculation" to accept function and use multi-value variables

### DISCUSSION THREAD

**Discussion by deka** [grafana/grafana#38387] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

- I have a multi-value variable named "__apis__" (__value, __text)
- I use this variable to query data and filter to  api value like this :   
![image](https://user-images.githubusercontent.com/349156/84041265-e5e56180-a9a3-11ea-9087-993ced2ddc0c.png)
- I want to get a new column calculated with the description (__text) of the variable "apis" from its value (__value). New field is human readable. 

**Why is this needed**:

I think that the calculated fields should not be exclusively reserved for mathematical calculations

Thanks

PS : Is it understandable ? 😺

**Comment by torkelo** [comment]
no I do not understand what you want to do and how this relates to the calculation transform

**Comment by deka** [comment]
Ok @torkelo ,

I have a variable named 'apis' like this : 
| __value | __text |
|--------|--------|
| 1      | API 1  |
| 2      | API 2  |
| ...      | ...        |

I have data source (from elasticsearch) like this : 
| api | Count |
|-----|-------|
| 1   | 23    |
| 2   | 35    |
| ... | ...   |

After transformation i need something like : 
| api | api_name | Count |
|-----|----------|-------|
| 1   | API 1    | 23    |
| 2   | API 2    | 35    |
| ... |   ...    | ...   |


I think adding the 'api_name' column can be done with a calculated column. 
To do this, it is necessary that the transformation 'calculated column' accepts a function and that we can use the variable 'apis'

Something like for calculated column function: 
```
Function (row) {
  return  apis[row[api]] // return the __text from variable apis where __value == api column value
}
```

I don't know if we can implement this.

**Comment by deka** [comment]
News ?

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
